### PR TITLE
[Playground] [Backend] minor fixes for error msgs

### LINKED ISSUE #24030: [Playground] [Task]: Update error messages and http response for playground endpoints
### What needs to happen?

If a snippet is not found by ID should be HTTP code is 404 instead of 500

### Issue Priority

Priority: 3

### Issue Component

Component: beam-playground

### PR DESCRIPTION
**Pull request by vchunikhin**
fixes #24030
@rshamunov
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by vchunikhin** [comment]
R: @pabloem

**Comment by eantyshev** [comment]
LGTM

File: playground/backend/cmd/server/controller.go [hunk 1]
```diff
@@ -460,12 +463,15 @@ func (controller *playgroundController) SaveSnippet(ctx context.Context, req *pb
...
 	if controller.db == nil {
 		logger.Error("GetSnippet(): the runner is trying to read the snippet")
-		return nil, errors.InvalidArgumentError(errorTitleGetSnippet, "The runner doesn't support snippets")
+		return nil, cerrors.InvalidArgumentError(errorTitleGetSnippet, "The runner doesn't support snippets")
 	}
 	snippet, err := controller.db.GetSnippet(ctx, info.GetId())
 	if err != nil {
 		logger.Errorf("GetSnippet(): error during getting the snippet: %s", err.Error())
-		return nil, errors.InternalError(errorTitleGetSnippet, "Failed to retrieve the snippet")
+		if errors.Is(err, datastore.ErrNoSuchEntity) {
```
Comment (eantyshev) [inline playground/backend/cmd/server/controller.go:471]: that's the primary reason for this PR

File: .github/workflows/tour_of_beam_backend_integration.yml [hunk 2]
```diff
@@ -80,7 +80,7 @@ jobs:
             cache-read-only: false
 
       - name: Build Playground router image
-        run: ./gradlew playground:backend:containers:router:docker
```
Comment (eantyshev) [inline .github/workflows/tour_of_beam_backend_integration.yml:83]: this is to have more info in case of failure

**Review by damccorm — APPROVED** [review]
Thanks!

**Pull request merged**